### ISSUE TITLE
How to pick the best model during training?

### ISSUE THREAD

**Issue by undercutspiky** [eduardzamfir/seemoredetails#19]
Hi,

I'd like to reproduce your results for my research but looking at the code, it seems like you are saving the model every few checkpoints but I couldn't find anywhere how you're actually picking the best model. I'm guessing it's based on PSNR on the validation set.
But as per the yaml file, it seems like you're using Set5 as the validation set. I thought it was one of the test sets (as per the paper) and assumed that you'd use DIV2K's validation split as your validation set for tuning hyperparameters and picking the best checkpoint. Could you please clarify? 

Thank you.

https://github.com/eduardzamfir/seemoredetails/blob/825ba21f4a79323c5b847e86d31f8a0acbca287b/options/super_resolution/train/train_SeemoRe_T_X4.yml#L31-L36

**Comment by eduardzamfir** [comment]
Hi,

there is no model selection happing besides selecting the last checkpoint when training is completed. The validation on Set5 during training is only used for tracking the current performance during training. Usually SR models tend to perform better with longer training, therefore we keep the number of iterations similar to prior work, such as [SRFormer](https://github.com/HVision-NKU/SRFormer) or [DAT](https://github.com/zhengchen1999/DAT).

**Comment by undercutspiky** [comment]
Thank you. I didn't know that.
One question about that, which set do you use to tune the hyperparameters then? Is it one of the test sets or the DIV2K validation set?

**Comment by eduardzamfir** [comment]
I also didn’t do hyperparameter tuning. All hyperparameter such as batch size, patch size, learning rate, optimizer etc. are equal to prior work. The lambda weight between L1 and FFT loss was set to 0.1. I didn’t check whether different weighting leads to better results.
Of course you may use DIV2K Val for that.

**Comment by undercutspiky** [comment]
Oh okay. But did you run some validation while designing the new architecture...like to evaluate various choices...number of MoE blocks and so on?

**Comment by eduardzamfir** [comment]
Yes, for that I always ran the full evaluations on the benchmarks. You can have a look at the ablation section in the paper.
